### PR TITLE
fix(platform): add error message for combobox component

### PR DESCRIPTION
**Pull request by OleksiyDrozdov**
#### Please provide a link to the associated issue.
Closes https://github.com/SAP/fundamental-ngx/issues/3277

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
- [x] Run npm run build-pack-library and test in external application

Documentation checklist:
- [n/a] update `README.md`
- [n/a] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [n/a] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

File: libs/platform/src/lib/components/form/combobox/combobox.module.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { CommonModule } from '@angular/common';
 import { FormsModule } from '@angular/forms';
 import { OverlayModule } from '@angular/cdk/overlay';
 
-import { ButtonModule, DynamicComponentService, InputGroupModule, ListModule, TemplateModule } from '@fundamental-ngx/core';
+import { ButtonModule, DynamicComponentService, InputGroupModule, ListModule, TemplateModule, FormModule } from '@fundamental-ngx/core';
```
Comment (N1XUS) [inline libs/platform/src/lib/components/form/combobox/combobox.module.ts:6]: Please update imports with deep imports like
`import { ButtonModule } from '@fundamental-ngx/core/button'`

File: libs/platform/src/lib/components/form/combobox/commons/base-combobox.ts [hunk 2]
```diff
@@ -179,7 +186,7 @@ export abstract class BaseCombobox extends CollectionBaseInput implements AfterV
     /** @hidden */
     @ContentChildren(TemplateDirective)
     customTemplates: QueryList<TemplateDirective>;
-
+    _flatSuggestions: SelectableOptionItem[];
```
Comment (N1XUS) [inline libs/platform/src/lib/components/form/combobox/commons/base-combobox.ts:189]: Description or hidden flag is missing

File: libs/platform/src/lib/components/form/combobox/combobox/combobox.component.ts [hunk 3]
```diff
@@ -58,9 +58,17 @@ export class ComboboxComponent extends BaseCombobox implements OnInit, AfterView
     /** @hidden */
     selected?: OptionItem;
```
Comment (InnaAtanasova) [inline libs/platform/src/lib/components/form/combobox/combobox/combobox.component.ts:59]: If this is hidden it should be prefixed with `_`, then there will be duplication with the prop on line 64

File: libs/platform/src/lib/components/form/combobox/combobox/combobox.component.ts [hunk 4]
```diff
@@ -253,4 +259,32 @@ export class ComboboxComponent extends BaseCombobox implements OnInit, AfterView
             }, [])
             .find(item => item.label === text);
     }
+
+    validateOnKeyup(event: KeyboardEvent): void {
```
Comment (InnaAtanasova) [inline libs/platform/src/lib/components/form/combobox/combobox/combobox.component.ts:263]: missing typedocs

File: libs/platform/src/lib/components/form/combobox/commons/base-combobox.ts [hunk 5]
```diff
@@ -74,7 +75,10 @@ export abstract class BaseCombobox extends CollectionBaseInput implements AfterV
     /** Provides maximum height for the optionPanel */
     @Input()
     maxHeight = '250px';
-
+    /** @hidden */
+    isListEmpty = true;
+    /** @hidden */
+    isSearchInvalid = false;
```
Comment (InnaAtanasova) [inline libs/platform/src/lib/components/form/combobox/commons/base-combobox.ts:81]: Hidden props and methods are prefixed

**Comment by DeepakSap14** [comment]
@OleksiyDrozdov 
we already have infrastructure to display error message.

If we provide error message template then form-group takes care of displaying error message.
please see https://github.com/SAP/fundamental-ngx/blob/7db770e85cb62b0fecec6247c52b5ac2422af75e/apps/docs/src/app/platform/component-docs/platform-forms/platform-date-picker/platform-date-picker-examples/platform-date-picker-example.component.html#L34.

please see if this resolves the issue here.

**Comment by OleksiyDrozdov** [comment]
> @OleksiyDrozdov
> we already have infrastructure to display error message.
> 
> If we provide error message template then form-group takes care of displaying error message.
> please see
> 
> https://github.com/SAP/fundamental-ngx/blob/7db770e85cb62b0fecec6247c52b5ac2422af75e/apps/docs/src/app/platform/component-docs/platform-forms/platform-date-picker/platform-date-picker-examples/platform-date-picker-example.component.html#L34
> 
> .
> please see if this resolves the issue here.

@DeepakSap14 
I tried suggested implementation with error message template and it is not working. Error pop up is not shown.
Could you please advise if it is ok to leave previous implementation way that is working.

**Comment by DeepakSap14** [comment]
> > @OleksiyDrozdov
> > we already have infrastructure to display error message.
> > If we provide error message template then form-group takes care of displaying error message.
> > please see
> > https://github.com/SAP/fundamental-ngx/blob/7db770e85cb62b0fecec6247c52b5ac2422af75e/apps/docs/src/app/platform/component-docs/platform-forms/platform-date-picker/platform-date-picker-examples/platform-date-picker-example.component.html#L34
> > 
> > .
> > please see if this resolves the issue here.
> 
> @DeepakSap14
> I tried suggested implementation with error message template and it is not working. Error pop up is not shown.
> Could you please advise if it is ok to leave previous implementation way that is working.

@OleksiyDrozdov 
my mentioned case will work, when combobox used inside fdp-form-group. there we can pass custom validator to validate and on failure error message will be shown.
but combobox can also be used outside form as well, and for that i think your implementation is correct way for showing error message.

File: libs/platform/src/lib/components/form/combobox/combobox.module.ts [hunk 6]
```diff
@@ -3,7 +3,11 @@ import { CommonModule } from '@angular/common';
 import { FormsModule } from '@angular/forms';
 import { OverlayModule } from '@angular/cdk/overlay';
 
-import { ButtonModule, DynamicComponentService, InputGroupModule, ListModule, TemplateModule } from '@fundamental-ngx/core';
+import { ButtonModule } from '@fundamental-ngx/core/button'; 
+import { DynamicComponentService, TemplateModule } from '@fundamental-ngx/core/utils';
+import { InputGroupModule } from '@fundamental-ngx/core/input-group';
+import { ListModule } from '@fundamental-ngx/core/list';
+import { FormModule } from '@fundamental-ngx/core';
```
Comment (InnaAtanasova) [inline libs/platform/src/lib/components/form/combobox/combobox.module.ts:10]: import { FormModule } from '@fundamental-ngx/core/form';

**Review by KevinOkamoto — APPROVED** [review]

**Review by DeepakSap14 — APPROVED** [review]

**Review by InnaAtanasova — APPROVED** [review]

**Pull request merged**